src/runtime/runtime.cs

```diff
@@ -1893,25 +1908,24 @@ internal static IntPtr PyMem_Realloc(IntPtr ptr, long size)
 
         internal static void SetNoSiteFlag()
         {
+            if (_PythonDll == "__Internal")
+            {
+                throw new NotSupportedException("SetNoSiteFlag didn't support on static compile");
+            }
             var loader = LibraryLoader.Get(OperatingSystem);
-
-            IntPtr dllLocal;
-            if (_PythonDll != "__Internal")
+            IntPtr dllLocal = loader.Load(_PythonDll);
+            if (dllLocal == IntPtr.Zero)
             {
-                dllLocal = loader.Load(_PythonDll);
+                throw new Exception($"Cannot load {_PythonDll}");
```

Comment:
Can you throw more informative exception? `Win32Exception` would automatically read last error on Windows, but I don't know what to do for *nix

Reply:
Generally it should throw inside from `Load`, the nullptr checking just for in case(if the loader implementation didn't throw an exception). I'm not going to insert any platform specific code here.